data_fetcher.py

```diff
@@ -278,63 +302,82 @@ def is_subcircle(circle):
     return modified_plan
 
 
-
 async def process_req_plan(req_dataset, req_create_lyr):
     action = req_create_lyr.action
     plan: List[str] = []
     current_plan_index = 0
 
     if (
-        req_dataset.radius > 750
-        and req_dataset.page_token == ""
+        req_dataset.page_token == ""
         and action == "full data"
     ):
-        circle_hierarchy = cover_circle_with_seven_circles(
-            (req_dataset.lng, req_dataset.lat), req_dataset.radius / 1000
-        )
-        type_string = make_include_exclude_name(
-            req_dataset.includedTypes, req_dataset.excludedTypes
-        )
-        string_list_plan = create_string_list(
-            circle_hierarchy, type_string, req_dataset.text_search
-        )
-        string_list_plan.append("end of search plan")
+    
+    
+        if isinstance(req_dataset, ReqRealEstate) :
+            string_list_plan=await create_real_estate_plan(req_dataset)
+            string_list_plan.append("end of search plan")
+
+        if isinstance(req_dataset, ReqLocation) : 
+            circle_hierarchy = cover_circle_with_seven_circles(
+                (req_dataset.lng, req_dataset.lat), req_dataset.radius / 1000
+            )
+            type_string = make_include_exclude_name(
+                req_dataset.includedTypes, req_dataset.excludedTypes
+            )
+            string_list_plan = create_string_list(
+                circle_hierarchy, type_string, req_dataset.text_search
+            )
+            string_list_plan.append("end of search plan")

```

Comment:
let's try to not repeat lines of code we don't need to , 

```suggestion
        if isinstance(req_dataset, ReqRealEstate) :
            string_list_plan=await create_real_estate_plan(req_dataset)

        if isinstance(req_dataset, ReqLocation) : 
            circle_hierarchy = cover_circle_with_seven_circles(
                (req_dataset.lng, req_dataset.lat), req_dataset.radius / 1000
            )
            type_string = make_include_exclude_name(
                req_dataset.includedTypes, req_dataset.excludedTypes
            )
            string_list_plan = create_string_list(
                circle_hierarchy, type_string, req_dataset.text_search
            )
        
        
        string_list_plan.append("end of search plan")

```